asm/instruction.go

```diff
@@ -356,12 +398,34 @@ func (ins *Instruction) copyMetadata() *metadata {
 	return ins.metadata
 }
 
+// setMap sets the *ebpf.Map from which this instruction preforms a data.
+func (ins *Instruction) setMap(m FDer) {
```

Comment:
We could by embedding `fmt.Stringer` in the `FDer` interface, since `*ebpf.Map` does have a [String()](https://github.com/cilium/ebpf/blob/d4100a1f430f7792dd6ab3113a2dbbe6a1c10e82/map.go#L461) func.

We currently already have the reference as `<ref>` behind instructions, so after replacing a string reference with a map we could replace it with the string'ed version of the map(its type, name and fd). So that seems logical.

On the other hand, I would rather do that in a separate PR to avoid feature creep in this PR, otherwise it will never be merged :smiley:

Reply:
Yep, let's do this in a follow-up. I agree it's a must-have, though.

Reply:
SGTM. We don't need fmt.Stringer in `FDer`, calling `fmt.Sprint` is enough and will figure out the correct way to stringify.

Reply:
I was in the neighborhood anyway to remove `MapPtr()`, added this anyway in the PR, wasn't that much work